File: src/jvmception/objects/cp/MethodInfo.java
package jvmception.objects.cp;

import java.io.DataInputStream;
import java.io.IOException;

public class MethodInfo extends FieldInfo{
	
	private CodeAttribute code = null;
	
	public MethodInfo(DataInputStream dis, CpInfo[] cpInfo) throws IOException {
		super(dis, cpInfo);
		
		for (int i = 0; i < this.getAttributes().length; i++) {
			if (this.getAttributes()[i] instanceof CodeAttribute){
				code = (CodeAttribute) this.getAttributes()[i];
			}
		}
	}

	public CodeAttribute getCodeAttribute() {
		return code;
	}
}

File: src/jvmception/objects/cp/MethodInfo.cs
using System.IO;

public class MethodInfo : FieldInfo
{

    private CodeAttribute code = null;

    public MethodInfo(BinaryReaderBigEndian dis, CpInfo[] cpInfo) : base(dis, cpInfo)
    {
        for (int i = 0; i < this.getAttributes().Length; i++)
        {
            if (this.getAttributes()[i] is CodeAttribute)
            {
                code = (CodeAttribute)this.getAttributes()[i];
            }
        }
    }

    internal MethodInfo(string name, string descriptor) : base(name, descriptor)
    {

        code = new CodeAttribute();
    }

    public CodeAttribute getCodeAttribute()
    {
        return code;
    }
}
